### PR TITLE
remove install script and use json

### PR DESCRIPTION
**Pull request by gene-redpanda**
This PR makes the necessary changes to remove the install script from the repo in support of the air gap project while replacing ever more static strings with variables users can set to improve their ability to control the install process. In addition I've swapped us from YAML to JSON for the templates so that we're better able to modularly include/exclude elements without running into mapping issues.
  
  - Bash Install Script Removal for RPM/DEB (Ansible changes)
    - [x] Eliminate bash install script for deb
    - [x] Eliminate bash install script for rpm
    - [x] Ensure there is a var to pass in the location of the install files as either a path or a hyperlink
    - [x] Split permissions for data directory from creation of data directory
    - [x] Switch templates from YAML to JSON for ease of future modification

### CODE REVIEW

**Comment by gene-redpanda** [comment]
related to https://github.com/redpanda-data/deployment-automation/issues/176

File: roles/redpanda_broker/defaults/main.yml [hunk 1]
```diff
@@ -26,3 +29,25 @@ node_exporter_version: "{{ node_exporter_custom_version | default('1.5.0') }}"
 cloud_storage_credentials_source: "config_file"
 cloud_storage_enable_remote_write: "true"
 cloud_storage_enable_remote_read: "true"
+rp_key_rpm: "https://dl.redpanda.com/public/redpanda/gpg.988A7B0A4918BC85.key"
+rp_config_url_rpm: "https://dl.redpanda.com/public/redpanda/config.rpm.txt"
+rp_conf_loc_rpm: "/etc/yum.repos.d/redpanda-redpanda.repo"
+rp_key_deb: "https://dl.redpanda.com/sMIXnoa7DK12JW4A/redpanda/gpg.988A7B0A4918BC85.key"
+rp_config_url_deb: "https://dl.redpanda.com/sMIXnoa7DK12JW4A/redpanda/config.deb.txt"
```
Comment (hcoyote) [inline roles/redpanda_broker/defaults/main.yml:36]: given we're doing this in part for better airgap support, we should break these urls into a host and path component so the user only has to modify one variable.

also: the gpg keys are the same I think, so we could probably reduce this down to one rp_key variable. 

not sure why the paths would be different anyway (public vs sMIXnoa7DK12JW4A)

File: roles/redpanda_broker/tasks/install-redpanda.yml [hunk 2]
```diff
@@ -1,17 +1,50 @@
 ---
-- name: Download the Redpanda repo script
-  ansible.builtin.get_url:
-    url: "{{ redpanda_repo_script }}"
-    dest: "/tmp/redpanda_repo_script.sh"
-    mode: '0755'
+- name: Download and import Redpanda GPG Key
+  shell: |
+    curl -1sLf "{{ rp_key_deb }}" | gpg --dearmor > {{ rp_key_path_deb }}
+  when: ansible_os_family == 'Debian'
```
[the remark this answers is not in the document]
Comment (gene-redpanda) [reply roles/redpanda_broker/tasks/install-redpanda.yml:5]: I don't love it, but if you walk the commit history you can see I tried other solutions without success. I'm going to come back to it again in phase 3.

File: roles/redpanda_broker/templates/configs/defaults.j2 [hunk 3]
```diff
@@ -1,55 +1,78 @@
-cluster:
```
[the remark this answers is not in the document]
Comment (gene-redpanda) [reply roles/redpanda_broker/templates/configs/defaults.j2:1]: It's not essential for this PR and I suppose I could always pull it out. That said I kind of consider this another major version jump either way so I'm not super concerned about backwards compatibility.

File: roles/redpanda_broker/tasks/install-redpanda.yml [hunk 4]
```diff
@@ -1,17 +1,50 @@
...
+    update_cache: yes
+  when: ansible_os_family == 'Debian'
+
+- name: Add Redpanda repository (source)
+  apt_repository:
+    filename: "redpanda-redpanda-src"
+    mode: 0644
+    repo: "{{ rp_repo_signing_src_deb }}"
+    update_cache: yes
+  when: ansible_os_family == 'Debian'
```
[the remark this answers is not in the document]
Comment (hcoyote) [reply roles/redpanda_broker/tasks/install-redpanda.yml:21]: Fair enough. If we need to, we can gate it behind a variable.

File: roles/redpanda_broker/defaults/main.yml (hunk 1 above)
Comment (gene-redpanda) [reply roles/redpanda_broker/defaults/main.yml]: I'm not sure either so I went with Chesterton's fence and kept both keys LOL

I've just pushed a commit that adds a base url element

File: roles/redpanda_broker/templates/configs/defaults.j2 (hunk 3 above)
[the remark this answers is not in the document]
Comment (hcoyote) [reply roles/redpanda_broker/templates/configs/defaults.j2:1]: I'd prefer we pull it out. Moving to directly managing the repo configs should still result in no real behavioral change to the user. It's implementation detail more than anything.

If I understand the json thing, the user would have to go change their ansible if they happen to be using their own additions to custom_config_template.

[the remark this answers is not in the document]
Comment (gene-redpanda) [reply roles/redpanda_broker/templates/configs/defaults.j2:1]: Well TIL custom_config_template and custom_config_template**s** are not the same thing: our templates. I'm going to double check that I can just leave that bit yaml

[the remark this answers is not in the document]
Comment (gene-redpanda) [reply roles/redpanda_broker/templates/configs/defaults.j2:1]: @hcoyote ok, so on review -- as opposed to hot take mode -- custom_config_template is just the loop_var for custom_config_templates. That means users can't* define a new value for it because it is scoped to the loop. 

Users are welcome to replace custom_config_templates, but that's an unsupported path. If they do it, they should fork the repo and carefully decide what they want to include.

The intended method for on the fly modification of config is passing JSON in through the redpanda variable. 

* I mean in theory they can't, but in practice it's python so 🤷‍♀️

File: roles/redpanda_broker/tasks/install-redpanda.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hcoyote) [reply roles/redpanda_broker/tasks/install-redpanda.yml:5]: Sounds good. Fine with this as is for now then.

The alternative is doing the curl part with native ansible builtins, and then just doing the gpg dearmor as a shell operation. Then you could gate the dearmor based on file creation time or something.

We should probably figure out if the dearmor is even necessary. I'm seeing conflicting advice on this and I'm not embedded enough in the debian way to know what's true.

File: roles/redpanda_broker/templates/configs/defaults.j2 (hunk 3 above)
[the remark this answers is not in the document]
Comment (hcoyote) [reply roles/redpanda_broker/templates/configs/defaults.j2:1]: I meant custom_config_templates, sorry!

File: roles/redpanda_broker/tasks/install-redpanda.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gene-redpanda) [reply roles/redpanda_broker/tasks/install-redpanda.yml:5]: I rejected ansible builtin curl because it required a tmp file to pass into dearmor and that felt ugly. I suppose I could always go that route. 

Beyond that from what I understand we can do either dearmor or rename to asc. Rename to asc caused key match to fail so I'm not sure whether that info is good or not, but I tested it pretty thoroughly and couldn't get it to work. Curl into dearmor isn't ideal because on second run it repeats.

**Comment by gene-redpanda** [comment]
@hcoyote re: https://github.com/redpanda-data/redpanda-ansible-collection/pull/14#discussion_r1200911361

this was a derp on my part. I forgot to put in the deb repo config stuff I had been working. It's all in now and that config pull is removed 🤦‍♀️

File: roles/redpanda_broker/tasks/install-redpanda.yml [hunk 5]
```diff
@@ -31,10 +47,14 @@
...
 
+- name: Set conf location based on OS family
+  set_fact:
+    conf_dest: "{{ rp_conf_loc_deb if ansible_os_family == 'Debian' else rp_conf_loc_rpm }}"
+
 - name: Fetch Redpanda configuration
   uri:
-    url: "{{ conf_url }}?distro={{ ansible_distribution }}&codename={{ ansible_distribution_release }}&version={{ ansible_distribution_version }}&arch={{ ansible_architecture }}"
-    dest: /etc/yum.repos.d/redpanda-redpanda.repo
+    url: "{{ (conf_url ~ '?distro=' ~ ansible_distribution ~ '&codename=' ~ ansible_distribution_release ~ '&version=' ~ ansible_distribution_version ~ '&arch=' ~ ansible_architecture) | replace(' ', '%20') | lower }}"
```
[the remark this answers is not in the document]
Comment (gene-redpanda) [reply roles/redpanda_broker/tasks/install-redpanda.yml]: rip this out of date stuff

File: roles/redpanda_broker/defaults/main.yml [hunk 6]
```diff
@@ -9,13 +9,18 @@ redpanda_kafka_port: 9092
 redpanda_rpc_port: 33145
 
 redpanda_use_staging_repo: false
-redpanda_repo_debian: "{{ 'https://packages.vectorized.io/sMIXnoa7DK12JW4A/redpanda/cfg/setup/bash.deb.sh' if not redpanda_use_staging_repo|bool else 'https://dl.redpanda.com/E4xN1tVe3Xy60GTx/redpanda-unstable/setup.deb.sh' }}"
-redpanda_repo_redhat: "{{ 'https://dl.redpanda.com/nzc4ZYQK3WRGd9sy/redpanda/cfg/setup/bash.rpm.sh' if not redpanda_use_staging_repo|bool else 'https://dl.redpanda.com/E4xN1tVe3Xy60GTx/redpanda-unstable/setup.rpm.sh' }}"
+redpanda_base_url: "https://dl.redpanda.com"
+
+redpanda_repo_debian: "{{ 'https://packages.vectorized.io/sMIXnoa7DK12JW4A/redpanda/cfg/setup/bash.deb.sh' if not redpanda_use_staging_repo|bool else '{{ redpanda_base_url }}/E4xN1tVe3Xy60GTx/redpanda-unstable/setup.deb.sh' }}"
+redpanda_repo_redhat: "{{ '{{ redpanda_base_url }}/nzc4ZYQK3WRGd9sy/redpanda/cfg/setup/bash.rpm.sh' if not redpanda_use_staging_repo|bool else '{{ redpanda_base_url }}/E4xN1tVe3Xy60GTx/redpanda-unstable/setup.rpm.sh' }}"
```
Comment (r-vasquez) [inline roles/redpanda_broker/defaults/main.yml:15]: [question]

1. Why Debian default to `packages.vectorized` and redhat uses `dl.redpanda` ? I thought we were migrating away from packages.vectorized.
2. Why using the unstable version?

File: roles/redpanda_broker/tasks/install-redpanda.yml [hunk 7]
```diff
@@ -1,17 +1,79 @@
...
+
+- name: Install GPG key (RPM)
+  ansible.builtin.rpm_key:
+    state: present
+    key: "{{ rp_key_rpm }}"
+  when: ansible_os_family == 'RedHat'
 
 - name: Install redpanda from repository
-  ansible.builtin.apt:
+  ansible.builtin.package:
```
Comment (r-vasquez) [inline roles/redpanda_broker/tasks/install-redpanda.yml:76]: Why the change? I'm curious.

`ansible.builtin.package` defaults to **use:auto** and detects which package manager should use, but we have a **when** clause checking for the OS family. Wonder if we are using this `package` maybe we can migrate both (L76 and L85)

File: roles/redpanda_broker/defaults/main.yml (hunk 6 above)
Comment (gene-redpanda) [reply roles/redpanda_broker/defaults/main.yml:15]: Mostly because that's what they were before and I haven't looked into what the correct answer would be.

File: roles/redpanda_broker/tasks/install-redpanda.yml (hunk 7 above)
Comment (gene-redpanda) [reply roles/redpanda_broker/tasks/install-redpanda.yml:76]: I had intended to convert the name block to accept a fact + eliminate the duplicate .package call but never got around to the fact.

File: roles/redpanda_broker/defaults/main.yml (hunk 6 above)
Comment (r-vasquez) [reply roles/redpanda_broker/defaults/main.yml:15]: Can we ask devprod and update the vectorized one please? I fear that they might drop that in favor of cloudsmith (dl.redpanda).

Maybe something for another PR, lmk if that's the case and I'll approve.

Comment (gene-redpanda) [reply roles/redpanda_broker/defaults/main.yml:15]: Asked https://redpandadata.slack.com/archives/C02LZGSS66M/p1685051234352679

Comment (gene-redpanda) [reply roles/redpanda_broker/defaults/main.yml]: Oh right I'm not using these anymore, deleted

File: roles/redpanda_broker/tasks/install-redpanda.yml (hunk 7 above)
Comment (gene-redpanda) [reply roles/redpanda_broker/tasks/install-redpanda.yml]: done

**Review by r-vasquez — APPROVED** [review]
LGTM, thanks!
